src/Components/Web/src/Forms/InputFile/InputFileChangeEventArgs.cs

```diff
@@ -11,18 +11,48 @@ namespace Microsoft.AspNetCore.Components.Forms
     /// </summary>
     public sealed class InputFileChangeEventArgs : EventArgs
     {
+        private readonly IReadOnlyList<IBrowserFile> _files;
+
         /// <summary>
         /// Constructs a new <see cref="InputFileChangeEventArgs"/> instance.
         /// </summary>
         /// <param name="files">The list of <see cref="IBrowserFile"/>.</param>
         public InputFileChangeEventArgs(IReadOnlyList<IBrowserFile> files)
         {
-            Files = files;
+            _files = files ?? throw new ArgumentNullException(nameof(files));
         }
 
         /// <summary>
-        /// Gets the file entries list.
+        /// Gets the number of supplied files.
+        /// </summary>
+        public int FileCount => _files.Count;
+
+        /// <summary>
+        /// Gets the supplied file, or null if no file was supplied. Note that if the input
+        /// accepts multiple files, then instead of reading this property, you should call
+        /// <see cref="AcceptMultipleFiles(int)"/>.
         /// </summary>
-        public IReadOnlyList<IBrowserFile> Files { get; init; }
+        public IBrowserFile? File => _files.Count switch
+        {
+            0 => null,
+            1 => _files[0],
+            _ => throw new InvalidOperationException($"More than one file was supplied. Call {nameof(AcceptMultipleFiles)} to receive multiple files."),
+        };
+
+        /// <summary>
+        /// Gets the file entries list. This method should be used for inputs that accept multiple
+        /// files. If the input accepts only a single file, then use the <see cref="File"/> property
+        /// instead.
+        /// </summary>
+        /// <param name="maximumFileCount">The maximum number of files to accept. If the number of files exceeds this value, this method will throw an exception.</param>
+        public IReadOnlyList<IBrowserFile> AcceptMultipleFiles(int maximumFileCount = 10)
```

Comment:
`GetFiles`? This feels like a weird name.

Reply:
Counter offer: `GetMultipleFiles`

I'm just trying to make it clear that you're only meant to use this if your scenario involves `multiple`.

Reply:
That sounds reasonable.